From ef1238e67f70a2099c53fc64274fad954a284cd4 Mon Sep 17 00:00:00 2001
From: Slawomir Rosek <slawomir.rosek@cavium.com>
Date: Tue, 9 Jan 2018 11:37:58 +0100
Subject: [PATCH 0365/1051] octeontx: update dependencies in Kconfig

This patch fixes incomplete dependecies between 83xx co-proc modules. It
was omitted during merge, because originally the change was provided
together with CPT patch series instead of being separated to another patch.

CC: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Signed-off-by: Slawomir Rosek <slawomir.rosek@cavium.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/cavium/Kconfig b/drivers/net/ethernet/cavium/Kconfig
index cd0d603fd49a..6cf7be85eb77 100644
--- a/drivers/net/ethernet/cavium/Kconfig
+++ b/drivers/net/ethernet/cavium/Kconfig
@@ -144,7 +144,7 @@ config OCTEONTX_SSOW_PF
 
 config OCTEONTX_PKO_PF
 	tristate "OcteonTX PKO physical function driver(PKO_PF)"
-	depends on 64BIT
+	depends on 64BIT && OCTEONTX_FPA_PF && OCTEONTX_FPA_VF
 	default y
 	help
 	  Select this option to enable PKO Physical function.
@@ -202,8 +202,8 @@ config OCTEONTX_SLI_PF
 config OCTEONTX
 	tristate "OcteonTX coprocessor maintanier"
 	depends on THUNDER_NIC_BGX && OCTEONTX_FPA_PF && OCTEONTX_SSO_PF
-	depends on OCTEONTX_PKO_PF
-	depends on OCTEONTX_DPI
+	depends on OCTEONTX_PKO_PF && OCTEONTX_SSOW_PF && OCTEONTX_PKI
+	depends on OCTEONTX_DPI && OCTEONTX_LBK && OCTEONTX_TIM_PF
 	depends on OCTEONTX_ZIP_PF
 	depends on OCTEONTX_SLI_PF
 	default y
-- 
2.17.1

